src/rpc/rawtransaction.cpp

```diff
@@ -413,8 +412,17 @@ CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniVal
         }
         outputs = std::move(outputs_dict);
     }
+
+    // Duplicate checking
+    std::set<CTxDestination> destinations;
+    bool has_data{false};
+
     for (const std::string& name_ : outputs.getKeys()) {
         if (name_ == "data") {
+            if (has_data) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, duplicate data");
```

Comment:
nit: "duplicate data" could be misleading, as it could refer to the key or the value.

Reply:
They are key-value pairs and neither the data key, nor the data value are allowed to be duplicate?

Reply:
Ok, fixed